src/main/java/com/aws/greengrass/clientdevices/auth/util/WildcardTrie.java

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.util;
+
+import com.aws.greengrass.authorization.AuthorizationHandler.ResourceLookupPolicy;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Copied from nucleus with some customizations for performance.
```

Comment:
what is different? Should it be merged into nucleus?

Reply:
Replaced concurrentmap with hashmap, comparing latest performance with previous seemed to have some improvement (not drastic though)

Reply:
removed the fork